### PR TITLE
Revamp configuration loading

### PR DESCRIPTION
**Pull request by mikz**
closes https://github.com/3scale/apicast/issues/230

# Breaking Change

Accidentally previous APIcast would try to lazy load configuration even in the production mode.

That means that newly started APIcast would fetch configuration for service 1.
When receiving request for service 1 it would use the downloaded configuration.
But when receiving configuration of non existing service, it would try to lazy lookup the configuration and it would access the remote API.

This is "fixed" in this branch and when APIcast is in boot mode it won't lazy load configuration, ever.

This could break @3scale/qe workflow when publishing production service and immediately trying production APIcast. Because it would not have to be reloaded in up to 5 minutes.

# Environment variables

This PR removes:
* **AUTO_UPDATE_INTERVAL**
* **AUTO_UPDATE_INTERVAL**
* **AUTO_UPDATE_INTERVAL**

And adds:
* **APICAST_CONFIGURATION_CACHE**
* **APICAST_CONFIGURATION_LOADER**

## APICAST_CONFIGURATION_LOADER

Can be set either to `boot` or `lazy`.
Boot will load configuration on boot or exit if there are any errors (but empty configuration will work).
Lazy will try to load configuration on incoming request.

## APICAST_CONFIGURATION_CACHE
Controls for how long (in seconds) the configuration will be stored. Can be set to 0 for not caching it at all.
Default is empty value which will cache it forever.

### CODE REVIEW

File: apicast/src/configuration_store.lua [hunk 1]
```diff
@@ -16,9 +15,18 @@ local mt = { __index = _M }
 
 function _M.new()
   return setmetatable({
+    -- services hashed by id, example: {
+    --   ["16"] = service1
+    -- }
```
Comment (mayorova) [inline apicast/src/configuration_store.lua]: This kind of comments are very helpful :) 👍

Comment (mikz) [inline apicast/src/configuration_store.lua]: Yep. Wanted to do it just for sake of having it written when I was changing the data structure :)
Will try to do that more in complicated cases.

**Comment by jcechace** [comment]
Generally speaking we should be fine with the configuration not being immediate as long as there is a mechanism to trigger the immediate reload manually -- waiting up to 5 minutes for the configuration to be reloaded doesn't really make sense in tests. 

I will let @mijaros confirm this since I'm still a bit out of picture after rejoining the team. though

**Comment by mikz** [comment]
Ok. We could:

* trigger reload by calling `POST /boot` on the management api
* set lower refresh rate
* trigger deploy on OpenShift

**Pull request merged**